cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -205,8 +207,8 @@ class device_row_comparator {
         }
 
         // Non-empty structs have been modified to only have 1 child when using this.
-        lcol = lcol.children()[0];
-        rcol = rcol.children()[0];
+        lcol = detail::structs_column_device_view(lcol).sliced_child(0);
+        rcol = detail::structs_column_device_view(rcol).sliced_child(0);
```

Comment:
This may not be correct, as I discovered that it gives wrong results with sliced input. This patch fixes it (https://github.com/rapidsai/cudf/pull/10667/files#diff-0743b70282a92785dee8e55adc4099f501b4ed6c20c21e39fe187b2f4f0664a0L560-R562). I didn't test with lists thus I'm not sure.

Reply:
@devavret already fixed that here: https://github.com/rapidsai/cudf/pull/10641/commits/023ec1f57cdd5ec79efbddba4f16d71996a5621f

Reply:
I meant that fix may still be wrong.

Reply:
Alright, just verified and confirm that the fix in this PR works for struct type. List type was not tested yet, but I assume that it is also supported correctly here. Will report any issue if I encounter.